### ISSUE TITLE
Enhancements to Inventory Auto-hide

### ISSUE THREAD

**Issue by skybaks** [XertroV/tm-editor-ui-toolbox#2]
I really like the inventory auto-hide feature you have implemented but I wonder if it might be possible to enhance it so that it would autofocus your inventory to a specific folder level when you select an item as described in https://github.com/skybaks/tm-editor-helpers/issues/46
I found that we should be able to programmatically press the folder-up-one-level button but I'm not sure if there is a way we could programmatically left-click on a folder or get it to focus to a level.

**Comment by XertroV** [comment]
> would autofocus your inventory to a specific folder level when you select an item 

> I found that we should be able to programmatically press the folder-up-one-level button but I'm not sure if there is a way we could programmatically left-click on a folder or get it to focus to a level.

We can 'left click' (select it), I added this to editor++ a few days ago. It is *necessary* that the inventory is using the right root node, though (else crash). 

WRT autofocus, not sure. I'll have a look today. There are a few functions here that I don't know what they do yet: https://next.openplanet.dev/Game/CGameEditorGenericInventory

**Comment by XertroV** [comment]
so .EnterDirectory() will go up one left based on whatever is hovered

before:
![image](https://github.com/XertroV/tm-editor-ui-toolbox/assets/1046448/18da7b88-8fd3-457e-88a7-2d362ce602a4)
after:
![image](https://github.com/XertroV/tm-editor-ui-toolbox/assets/1046448/349a3d0c-c8e8-4d59-ad04-57e3d224b427)

.CurrentSelectedNode is the highest green folder/article (folders select on hover, articles on click)
.CurrentDirectroy is the highest green folder

.OpenDirectory and .SelectNode both work for picking directories.

.SelectBrotherNode selects the sibling on the left/right (loops around between first/last)


The .GetArticle and .GetDirectory methods presumably return the node before/after (left/right of) the argument node. 

No way to focus a folder it seems.

**Comment by skybaks** [comment]
Yeah, sorry i wrote "left click" but i really meant "right click". Ive
looked through those inventory functions before and there doesnt seem to be
one that does the focus to a level, as you also point out

**Comment by skybaks** [comment]
Ok what do you think the viability might be on using our Interface access to hide the lower rows and then change the position of the upper rows?

**Comment by XertroV** [comment]
> Ok what do you think the viability might be on using our Interface access to hide the lower rows and then change the position of the upper rows?

Hmm, interesting idea. Could work.

Another idea I had was to experiment with actually removing stuff from the inventory and see if that changes the UI. mb very hacky but could just change 1 reference each time. and that assumes we can edit those references at all.
